tests-ng/test_basics.py

```diff
@@ -84,13 +87,13 @@ def test_kernel_not_tainted():
 @pytest.mark.booted(reason="Systemctl needs a booted system")
 def test_no_failed_units(systemd: Systemd, shell: ShellRunner):
     system_run_state = systemd.wait_is_system_running()
-    print(
+    logger.debug(
         f"Waiting for systemd to report the system is running took {system_run_state.elapsed_time:.2f} seconds, with state '{system_run_state.state}' and return code '{system_run_state.returncode}'."
     )
     failed_systemd_units = systemd.list_failed_units()
     for u in failed_systemd_units:
-        print(f"FAILED UNIT: {u}")
+        logger.debug(f"FAILED UNIT: {u}")
         shell(f"journalctl --unit {u.unit}")
     assert (
         not failed_systemd_units
-    ), f"{len(failed_systemd_units)} systemd units failed to load"
+    ), f"{failed_systemd_units} systemd units failed to load"
```

Comment:
Readability could be improved with: f"{len(failed_systemd_units)} systemd units failed to load: {failed_systemd_units!r}"

Above change would create output similar to `["network-online.service", "ceph.service"] systemd units failed to load` while the suggested one `2 systemd units failed to load: ["network-online.service", "ceph.service"]`.